core/tests/protractor_utils/ExplorationEditorHistoryTab.js

```diff
@@ -58,71 +58,68 @@ var ExplorationEditorHistoryTab = function() {
 
   this.getHistoryGraph = function() {
     return {
-      openStateHistory: function(stateName) {
-        stateNodes.map(function(stateElement) {
-          return stateNodeLabel(stateElement).getText();
-        }).then(function(listOfNames) {
-          var matched = false;
-          for (var i = 0; i < listOfNames.length; i++) {
-            if (listOfNames[i] === stateName) {
-              stateNodes.get(i).click();
-              matched = true;
-            }
-          }
-          if (!matched) {
-            throw new Error(
-              'State ' + stateName + ' not found by getHistoryGraph.');
-          }
+      openStateHistory: async function(stateName) {
+        var listOfNames = await stateNodes.map(async function(stateElement) {
+          return await stateNodeLabel(stateElement).getText();
         });
+        var matched = false;
+        for (var i = 0; i < listOfNames.length; i++) {
+          if (listOfNames[i] === stateName) {
+            await stateNodes.get(i).click();
+            matched = true;
+          }
+        }
+        if (!matched) {
+          throw new Error(
+            'State ' + stateName + ' not found by getHistoryGraph.');
+        }
       },
-      closeStateHistory: function() {
-        waitFor.elementToBeClickable(
+      closeStateHistory: async function() {
+        await waitFor.elementToBeClickable(
           closeStateHistoryButton,
           'Close State History button is not clickable');
-        expect(closeStateHistoryButton.isDisplayed()).toBe(true);
-        closeStateHistoryButton.click();
-        waitFor.invisibilityOf(
+        expect(await closeStateHistoryButton.isDisplayed()).toBe(true);
+        await closeStateHistoryButton.click();
+        await waitFor.invisibilityOf(
           closeStateHistoryButton,
           'Close State History button takes too long to disappear.');
       },
-      deselectTwoVersions: function(versionNumber1, versionNumber2) {
+      deselectTwoVersions: async function(versionNumber1, versionNumber2) {
         // Array starts at 0.
-        historyCheckboxSelector.count().then(function(totalVersionNumber) {
-          var v1Position = totalVersionNumber - versionNumber1;
-          var v2Position = totalVersionNumber - versionNumber2;
+        var totalVersionNumber = await historyCheckboxSelector.count();
+        var v1Position = totalVersionNumber - versionNumber1;
+        var v2Position = totalVersionNumber - versionNumber2;
 
-          expect(historyCheckboxSelector.get(v1Position).isDisplayed())
-            .toBe(true);
-          historyCheckboxSelector.get(v1Position).click();
+        expect(await historyCheckboxSelector.get(v1Position).isDisplayed())
+          .toBe(true);
+        await historyCheckboxSelector.get(v1Position).click();
```

Comment:
Need nested `await` because `historyCheckboxSelector` is an `ElementArrayFinder`